src/rebar_file_utils.erl

```diff
@@ -72,11 +72,14 @@ consult_config(State, Filename) ->
         [T] -> T;
         [] -> []
     end,
-    SubConfigs = [consult_config(State, Entry ++ ".config") ||
-                  Entry <- Config, is_list(Entry)
-                 ],
-
-    [Config | lists:merge(SubConfigs)].
+    lists:flatmap(
+      fun (SubConfig) when is_list(SubConfig) ->
+        case consult_config(State, SubConfig) of
+          [] -> consult_config(State, SubConfig ++ ".config");
```

Comment:
This merely mirrors an OTP's convention that ".config" suffix MAY be omitted. I'm not sure if people use "file.config" to specify "file.config.config", and I'll need to test if that actually works or not. (I sure hope they don't).